reactioncommerce/plugin-example
# feat: enabling typescript

Impact: **minor**
Type: **feature**

The template is typescript ready now. Also added a couple of scripts to watch and build.

## Breaking changes
None

## Testing
1. Clone the repo/branch, run `npm build:watch`. Write a `console.log` statement in `src/index.ts`.
2. Check that the files are created in the `dist/` folder.
3. Modify any of the files, check the files in `dist/` folder are updated.
4. From the `reaction-development-platform/reaction`, link it to the reaction api using `bin/package-link @reactioncommerce/api-plugin-example ../api-plugin-example`. 
5. Add this to `reaction/plugins.json` file by appending: `"example": "@reactioncommerce/api-plugin-example"` at the end.
6. Verify that the `console.log` statement added before gets printed.
